Refactor log analysis

sinistro14 opened the pull request.

sinistro14 committed:
Added Nextcloud API

sinistro14 committed:
Add activity_fetcher

sinistro14 committed:
Major refactor/renaming

sinistro14 committed:
Refactor configuration

sinistro14 committed:
Refactor dependencies

Kronostheus reviewed (changes requested):
@sinistro14 indicated changes still to be made

sinistro14 committed:
Refactor project config

Separated test configs from default ones.
Default configuration is now overwritten by app deploy configuration.
